extensions/emmet/src/abbreviationActions.ts

```diff
@@ -435,17 +442,32 @@ function expandAbbr(input: ExpandAbbreviationInput): string | undefined {
 
 	try {
 		// Expand the abbreviation
-		let expandedText = helper.expandAbbreviation(input.abbreviation, expandOptions);
+		let expandedText;
 
 		if (input.textToWrap) {
+			let parsedAbbr = helper.parseAbbreviation(input.abbreviation, expandOptions);
+			if (input.rangeToReplace.isSingleLine) {
+
+				// Fetch innermost element in the expanded abbreviation
+				let lastNode = parsedAbbr;
```

Comment:
lets rename this to `innerMostNode`. It will then match with your comment and is also more intuitive than `lastNode`